### PR TITLE
fix fatal error on media comment while activity stream is disabled  #1028

### PR DESCRIPTION
**Pull request by thrijith**
fix fatal error on media comment while activity stream is disabled  #1028

when activity stream is disabled  [`bp_activity_add`](https://github.com/rtMediaWP/rtMedia/blob/d9f1c7a17469d0e9113df81b0a765f3d8b472806/app/main/controllers/activity/RTMediaBuddyPressActivity.php#L693) and [`bp_activity_delete`](https://github.com/rtMediaWP/rtMedia/blob/d9f1c7a17469d0e9113df81b0a765f3d8b472806/app/main/controllers/activity/RTMediaBuddyPressActivity.php#L747)  will not work so added condition to check if the function is available to solve the fatal error.

### CODE REVIEW

**Comment by kjnanda** [comment]
These options must not be visible if the `Activity Component` is disabled.

screenshot-krupa rtdemo in 2018-01-02 12-26-52-873

CC @pranali333 Please add your input.

File: app/admin/RTMediaFormHandler.php [hunk 1]
```diff
@@ -1153,20 +1156,42 @@ static function buddypress_render_options( $options ) {
...
 
 		do_action( 'rtmedia_buddypress_setting_content' );
+		/**
+		 * Disable inputs and change background color to differentiate disabled inputs.
+		 * If activity stream is disabled in BuddyPress Settings.
+		*/
+		if ( ! bp_is_active( 'activity' ) ) {
+			?>
+			<script>
+				jQuery( '#rtmedia-bp-enable-activity' ).prop( "disabled", true ).next().css( 'background', 'grey' );
```
Comment (BhargavBhandari90) [inline app/admin/RTMediaFormHandler.php]: Use hexcode instead of `grey`.

File: app/admin/RTMediaFormHandler.php [hunk 2]
```diff
@@ -1153,20 +1156,42 @@ static function buddypress_render_options( $options ) {
...
+		 * Disable inputs and change background color to differentiate disabled inputs.
+		 * If activity stream is disabled in BuddyPress Settings.
+		*/
+		if ( ! bp_is_active( 'activity' ) ) {
+			?>
+			<script>
+				jQuery( '#rtmedia-bp-enable-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );
+				jQuery( '#rtmedia-activity-feed-limit' ).prop( "disabled", true );
+				jQuery( '#rtmedia-enable-like-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );
+				jQuery( '#rtmedia-enable-comment-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );
```
Comment (BhargavBhandari90) [inline app/admin/RTMediaFormHandler.php]: You can make this one lined
`jQuery( '#rtmedia-bp-enable-activity, #rtmedia-enable-like-activity, #rtmedia-enable-comment-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );`

Sorry for not pointing out in previous review

File: app/admin/RTMediaFormHandler.php [hunk 3]
```diff
@@ -1153,20 +1157,53 @@ static function buddypress_render_options( $options ) {
...
+		$general_group      = apply_filters( 'rtmedia_buddypress_setting_group', $general_group );
+
+		$render_options = self::buddypress_render_options( $rtmedia->options );
+		// Change option description when activity stream is disabled.
+		if ( ! bp_is_active( 'activity' ) ) {
+			$render_options['rtmedia-enable-on-activity']['args']['desc']      =
+			$render_options['rtmedia-activity-feed-limit']['args']['desc']     =
+			$render_options['rtmedia-enable-like-activity']['args']['desc']    =
+			$render_options['rtmedia-enable-comment-activity']['args']['desc'] =
+				'Please Enable BuddyPress Activity Streams to update option';
```
Comment (BhargavBhandari90) [inline app/admin/RTMediaFormHandler.php]: Make this translatable

File: app/admin/RTMediaFormHandler.php [hunk 4]
```diff
@@ -1153,20 +1157,53 @@ static function buddypress_render_options( $options ) {
...
+		}
 
-		$render_options	= self::buddypress_render_options( $rtmedia->options );
-		$render_options	= apply_filters( 'rtmedia_album_control_setting', $render_options, $rtmedia->options );
+		if ( ! bp_is_active( 'groups' ) ) {
+			$render_options['rtmedia-enable-on-group']['args']['desc'] =
+				'Please Enable BuddyPress User Groups to update option';
```
Comment (BhargavBhandari90) [inline app/admin/RTMediaFormHandler.php]: Make this translatable

**Pull request merged**